### PR TITLE
Feat: Gnosis chain

### PR DESCRIPTION
**Pull request by Matthews3301**
# Description

Support for Gnosis Chain

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/services/balancer/pools/weighted-pool.service.ts [hunk 1]
```diff
@@ -38,6 +46,12 @@ export interface JoinPoolRequest {
 }
 
 export default class WeightedPoolService {
+  version: FactoryVersion;
+
+  constructor(public readonly _version: FactoryVersion) {
+    this.version = _version;
+  }
```
Comment (garethfuller) [inline src/services/balancer/pools/weighted-pool.service.ts:53]: When you define as `public readonly` it's automatically available on the instance. It's basically short hand for defining within the constructor.
```suggestion
  constructor(public readonly version: FactoryVersion) {}
```

File: src/services/balancer/pools/weighted-pool.service.ts [hunk 2]
```diff
@@ -48,29 +62,43 @@ export default class WeightedPoolService {
...
+            owner,
+          ]
+        : [
+            name,
+            symbol,
+            tokenAddresses,
+            seedTokens,
+            swapFeeScaled.toString(),
+            owner,
+          ];
```
Comment (garethfuller) [inline src/services/balancer/pools/weighted-pool.service.ts:96]: Does V3 not expect the same params as V2? Is it not only V1 that doesn't accept the rateProviders?

**Review by garethfuller — CHANGES_REQUESTED** [review]

File: src/services/balancer/pools/weighted-pool.service.ts (hunk 2 above)
Comment (Matthews3301) [reply src/services/balancer/pools/weighted-pool.service.ts:92]: Updated

**Comment by Matthews3301** [comment]
@garethfuller shall we merge this in? I'm still trying to verify to API, however, this codebase should work regardless.

Maybe removing the Gnosis-chain option from the dropdown but retaining url for testing?

**Comment by garethfuller** [comment]
@Matthews3301 no not yet, I'm still reviewing and I think we need to make some changes to the create a pool stuff.

**Comment by Matthews3301** [comment]
Continued in https://github.com/balancer-labs/frontend-v2/pull/2903/files

**Pull request closed without merging**